src/transport_xpmem.h

```diff
@@ -80,7 +80,7 @@ static inline
 int
 shmem_transport_xpmem_quiet(void)
 {
-    __sync_synchronize();
+    shmem_internal_membar();
```

Comment:
This looks like it will cause a circular include, since ```shmem_synchronization.h``` includes ```transport_xpmem.h``` (via ```shmem_comm.h``` -- our headers are such a mess).  You are missing the ```#include "shmem_synchronization.h"``` needed to define this function, but as soon as you include it, the compiler will barf (assuming you test with XPMEM enabled).  Better to leave this as ```__sync_synchrionize()```.

Reply:
```shmem_internal_membar()``` is defined in ```shmem_atomic.h```. I am not sure why ```shmem_synchronization.h``` would be needed here? Am I missing anything here?

Reply:
Ok, looks like I got my wires crossed.  I think you do need to ```#include "shmem_atomic.h"``` in ```transport_xpmem.h```.

Reply:
Ok. I will add it then. But, why dont I see the compiler complains about this in my local or in travis?

Reply:
It's probably already included in the file that includes transport_xpmem.h.